### PR TITLE
[X86] Remove Intel Xeon Phi Supports.

### PR DESCRIPTION
**Pull request by FreddyLeaf**
- [CFE][X86] Remove Xeon Phi CPU names supports.
- [X86][BE] Remove Xeaon Phi CPU names supports.
- [X86][BE] Workaround backend tests to work.
- [X86][BE] Workaround special tests to work.
- Remove AVX512ER intrinsic supports.
- Remove AVX512PF intrinsic supports.
- Remove PREFETCHWT1 intrinsic supports.

### CODE REVIEW

File: clang/lib/Headers/cpuid.h [hunk 1]
```diff
@@ -159,8 +159,6 @@
 #define bit_AVX512IFMA  0x00200000
 #define bit_CLFLUSHOPT  0x00800000
 #define bit_CLWB        0x01000000
-#define bit_AVX512PF    0x04000000
-#define bit_AVX512ER    0x08000000
```
Comment (RKSimon) [inline clang/lib/Headers/cpuid.h:163]: I'd probably keep these for future reference.

**Review by RKSimon — COMMENTED** [review]
I still think we should keep the knl/knm cpu models in place, even if they don't support all features anymore.

**Comment by RKSimon** [comment]
@FreddyLeaf Can this be abandoned now?

**Comment by FreddyLeaf** [comment]
> @FreddyLeaf Can this be abandoned now?

sure

**Pull request closed without merging**